consensus/src/aggregator.rs

```diff
@@ -321,20 +344,29 @@ mod tests {
             }
 
             // Check collected votes
-            let (_, quorum_reached) =
-                a.collect_vote(&c, sign_info, &vote, step).unwrap();
+            let (_, quorum_reached) = a.collect_vote(&c, msg).unwrap();
 
             assert!(!quorum_reached, "quorum should not be reached yet");
 
             collected_votes += expected_votes[i];
-            assert_eq!(a.get_total(step, vote.clone()), Some(collected_votes));
+            assert_eq!(a.get_total(msg.id(), msg.vote), Some(collected_votes));
 
-            // Ensure a duplicated vote is discarded
             if i == 0 {
-                match a.collect_vote(&c, sign_info, &vote, step) {
+                // Ensure a duplicated vote is discarded
+                match a.collect_vote(&c, msg) {
                     Err(AggregatorError::DuplicatedVote) => {}
                     _ => panic!("Vote should be discarded"),
                 }
+
+                // Ensure a conflicting vote is discarded
+                let mut wrong_msg = msg.clone();
+                wrong_msg.vote = Vote::Invalid(block_hash);
```

Comment:
Should the signature be updated accordingly?
This vote should be discarded due to invalid signature rather than being conflicting.

It seems that `is_vote_collected` is called before checking the signature. I think we should change this order

Reply:
Signature of any StepMessage is verified before any kind of process

So, when we collect it it's already verified